packages/zoe/src/contracts/callSpread.js

```diff
@@ -0,0 +1,217 @@
+// @ts-check
+import '../../exported';
+
+import { assert, details } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { E } from '@agoric/eventual-send';
+import {
+  assertProposalShape,
+  depositToSeat,
+  natSafeMath,
+  trade,
+  assertUsesNatMath,
+} from '../contractSupport';
+
+const { subtract, multiply, floorDivide } = natSafeMath;
+
+/**
+ * Constants for long and short positions.
+ *
+ * @type {{ LONG: 'long', SHORT: 'short' }}
+ */
+const Position = {
+  LONG: 'long',
+  SHORT: 'short',
+};
+
+const PERCENT_BASE = 100;
+const inverse = percent => subtract(PERCENT_BASE, percent);
+
+/**
+ * This contract implements a fully collateralized call spread. This is a
+ * combination of a call option bought at one strike price and a second call
```

Comment:
"One contract to rule them all" says "A more general structure is a call spread, which is a bought call at a lower strike and a sold call at a higher strike. This generalises calls, puts, binaries (including prediction markets), and futures."   It later says (under "Fully collateralized call spread") "This is a call at a lower strike and a put at a higher strike."

I think the second is a mistake. It would produce the right outcomes for the long position, but the short position wouldn't have all the rights they should. I've asked for clarification in the doc

Reply:
Joe responded that they are both calls. The earlier (in the document) statement was correct.